bindings/kitsune_client/src/op_store.rs

```diff
@@ -0,0 +1,379 @@
+use bytes::Bytes;
+use kitsune2::default_builder;
+use kitsune2_api::{
+    AgentId, BoxFut, Builder, Config, DhtArc, DynOpStore, K2Result, MetaOp, OpId, OpStore,
+    OpStoreFactory, SpaceId, Timestamp,
+};
+use std::sync::Arc;
+use wind_tunnel_instruments::{
+    prelude::{ReportMetric, Reporter},
+    OperationRecord,
+};
+
+#[derive(Debug)]
+pub(super) struct WtOpStoreFactory {
+    agent_id: AgentId,
+    reporter: Arc<Reporter>,
+}
+
+impl OpStoreFactory for WtOpStoreFactory {
+    fn create(
+        &self,
+        _builder: Arc<Builder>,
+        space: SpaceId,
+    ) -> BoxFut<'static, K2Result<DynOpStore>> {
+        let agent_id = self.agent_id.clone();
+        let reporter = self.reporter.clone();
+        Box::pin(async move {
+            // Create a memory op store. The passed in builder cannot be used for this, because
+            // it would call this create function recursively ad infinitum.
+            let builder = Arc::new(default_builder().with_default_config()?);
+            let op_store = builder.op_store.create(builder.clone(), space).await?;
+            let out: DynOpStore = Arc::new(WtOpStore::new(op_store, agent_id, reporter));
+            Ok(out)
+        })
+    }
+
+    fn default_config(&self, _config: &mut Config) -> K2Result<()> {
+        Ok(())
+    }
+
+    fn validate_config(&self, _config: &Config) -> K2Result<()> {
+        Ok(())
+    }
+}
+
+impl WtOpStoreFactory {
+    pub fn new(agent_id: AgentId, reporter: Arc<Reporter>) -> Self {
+        Self { agent_id, reporter }
+    }
+}
+
+#[derive(Debug)]
+struct WtOpStore {
+    inner: DynOpStore,
+    reporter: Arc<Reporter>,
+    agent_id: AgentId,
+}
+
+impl OpStore for WtOpStore {
+    fn process_incoming_ops(&self, op_list: Vec<Bytes>) -> BoxFut<'_, K2Result<Vec<OpId>>> {
+        Box::pin(async {
+            let inserted_op_ids = self.inner.process_incoming_ops(op_list).await?;
+            let amount = inserted_op_ids.len();
+            log::info!("{} ops have come in to {}", amount, self.agent_id);
+            if amount > 0 {
```

Comment:
`process_incoming_ops` is frequently called with 0 ops. I think that can be improved in kitsune's gossip protocol. For the time being only register an operation when ops have actually come in.